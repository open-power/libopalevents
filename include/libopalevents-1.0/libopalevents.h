/*
 *  libopalevents - OPAL Event parsing and printing library
 *  Copyright (C) 2015 IBM Corporation

 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.

 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.

 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

#ifndef _H_OPAL_EVENTS
#define _H_OPAL_EVENTS

#include <inttypes.h>

enum opal_event_log_flags {
	OPAL_EVENT_ALL = 0,
	OPAL_EVENT_ACTION = 1, /* Mutally exclusive */
	OPAL_EVENT_HEALTH = 2, /* Mutally exclusive */
	OPAL_EVENT_REPORT_ALL = 4,
	OPAL_EVENT_REPORT_HMC = 8,
	OPAL_EVENT_CALL_HOME = 16
};

typedef struct opal_event_log opal_event;

typedef enum opal_event_log_flags opal_event_flags;

opal_event *opal_event_create(void);

int opal_event_parse(opal_event *event, char *buf, int buflen);

int opal_event_print(opal_event *event);

int opal_event_summarise_hdr(void);

int opal_event_summarise(opal_event *event, opal_event_flags flags);

uint32_t opal_event_get_logid(opal_event *event);

int opal_event_destroy(opal_event *event);

#endif /* _H_OPAL_EVENTS */
