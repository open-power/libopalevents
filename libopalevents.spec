Name:           libopalevents
Version:        1
Release:        0%{?dist}
Summary:        PEL structure reader and parser

License:        LGPLv2
URL:            https://github.com/open-power/libopalevents
Source0:        https://github.com/open-power/libopalevents/archive/master.tar.gz
BuildRequires: autoconf automake libtool 
#Requires:       

%description 
PEL structure reader and parser
library to aid in reading, parsing and printing platform event log structures
 from files.



%package        devel
Summary:        Development files for %{name}
Requires:       %{name}%{?_isa} = %{version}-%{release}

%description    devel
The %{name}-devel package contains libraries and header files for
developing applications that use %{name}.


%prep

%setup -qn %{name}-master
autoreconf -if

%build
%configure --disable-static --prefix=/usr
make %{?_smp_mflags}


%install
rm -rf $RPM_BUILD_ROOT
%make_install
find $RPM_BUILD_ROOT -name '*.la' -exec rm -f {} ';'


%post -p /sbin/ldconfig

%postun -p /sbin/ldconfig


%files
%doc
%{_libdir}/*.so.*

%files devel
%doc
%{_includedir}/*
%{_libdir}/*.so
%{_libdir}/pkgconfig/libopalevents.pc


%changelog
* Wed Apr  8 2015 Cyril Bur <cyril.bur@au1.ibm.com> - 1
- Inital verion of the package
