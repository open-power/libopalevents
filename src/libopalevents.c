/*
 *  libopalevents - OPAL Event parsing and printing library
 *  Copyright (C) 2015 IBM Corporation

 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.

 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.

 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

/*
 *  libopalevents - OPAL Event parsing and printing library
 *  Copyright (C) 2015 IBM Corporation

 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.

 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.

 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */

#include <inttypes.h>

#include "libopalevents-1.0/libopalevents.h"
#include "opal-event-log.h"
#include "parse-opal-event.h"
#include "print-opal-event.h"

opal_event *opal_event_create(void)
{
	return create_opal_event_log();
}

int opal_event_parse(opal_event *event, char *buf, int buflen)
{
	return parse_opal_event_log(buf, buflen, event);
}

int opal_event_print(opal_event *event)
{
	return print_opal_event_log(event);
}

int opal_event_summarise_hdr(void)
{
	return header_opal_event_log();
}

int opal_event_summarise(opal_event *event, opal_event_flags flags)
{
	return summarise_opal_event_log(event, flags);
}

uint32_t opal_event_get_logid(opal_event *event)
{
	return opal_event_log_id(event);
}

int opal_event_destroy(opal_event *event)
{
	return free_opal_event_log(event);
}
