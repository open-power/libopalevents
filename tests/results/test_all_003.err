Warning parse_opal_event_log: detected missing section in opal event log
