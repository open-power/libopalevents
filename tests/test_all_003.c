#include <stdlib.h>
#include <fcntl.h>
#include <unistd.h>
#include "libopalevents-1.0/libopalevents.h"
#include "test.h"
#include "test_fns.h"

int main(int argc, char *argv[])
{
	TEST the_t = {
		.name = "test_all_003 (5032a000)",
		.t = &test_all_normal,
		.cookie = (void *) 16,
		.r = { "test_all_003.out", "test_all_003.err" },
		.i = { "test_all_003.raw" }
	};

	return test_main(the_t);
}
