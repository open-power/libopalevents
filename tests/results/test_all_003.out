|------------------------------------------------------------------------------|
|                                Private Header                                |
|------------------------------------------------------------------------------|
| Section Version          : 1 (PH)                                            |
| Sub-section type         : 0x0                                               |
| Section Length           : 0x30                                              |
| Component ID             : 2700                                              |
| Created at               : 2014-03-13 | 08:15:55                             |
| Committed at             : 2014-03-13 | 08:15:55                             |
| Created by               : Service Processor                                 |
| Creator Sub Id           : 0x0 (0), 0x0 (0)                                  |
| Platform Log Id          : 0x5034a000                                        |
| Entry ID                 : 0x5034a000                                        |
| Section Count            : 4                                                 |
|------------------------------------------------------------------------------|
|                                 User Header                                  |
|------------------------------------------------------------------------------|
| Section Version          : 1 (UH)                                            |
| Sub-section type         : 0x0                                               |
| Section Length           : 0x18                                              |
| Component ID             : 2700                                              |
| Subsystem                : Room ambient temperature                          |
| Event Scope              : Single platform                                   |
| Event Severity           : Predictive Error                                  |
| Event Type               : Not applicable.                                   |
| Action Flags             : Report to Operating System                        |
|                          : Service Action Required                           |
|------------------------------------------------------------------------------|
|                        Primary System Reference Code                         |
|------------------------------------------------------------------------------|
| Section Version          : 1 (PS)                                            |
| Sub-section type         : 0x1                                               |
| Section Length           : 0xa0                                              |
| Component ID             : 2700                                              |
| SRC Format               : 0x1                                               |
| SRC Version              : 0x2                                               |
| Valid Word Count         : 0x9                                               |
| SRC Length               : 98                                                |
| Primary Reference Code   : 11007201                                          |
| Hex Words 2 - 5          : 003C0001 00007201 00000000 00000000               |
| Hex Words 6 - 9          : 00000000 00000000 00000000 00000000               |
|                                                                              |
|                               Callout Section                                |
|                                                                              |
| Additional Sections      : Disabled                                          |
| Callout Count            : 1                                                 |
|                                                                              |
|                                 Symbolic FRU                                 |
| Priority                 : Mandatory, replace all with this type as a unit   |
| Location Code            : U78AB.001.WZSGBJ6                                 |
| Part Number              : AMBTEMP                                           |
| CCIN                     :                                                   |
| Serial Number            :                                                   |
| Machine Type Model       : 8246-L2C                                          |
| Serial Number            : 10008FA                                           |
| PCE                      :                                                   |
|                                                                              |
|------------------------------------------------------------------------------|
|                             Extended User Header                             |
|------------------------------------------------------------------------------|
| Version                  : 1 (EH)                                            |
| Sub-section type         : 0                                                 |
| Component ID             : 3100                                              |
| Machine Type Model       : 8246-L2C                                          |
| Serial Number            : 10008FA                                           |
| FW Released Ver          : ZL770_060                                         |
| FW SubSys Version        : b1212p_1320.770                                   |
| Common Ref Time (UTC)    :    0-00-00 | 00:00:00                             |
| Symptom Id Len           : 28                                                |
| Symptom Id               : 11007201_003C0001_00007201                        |
